### PR TITLE
Skip emitting constructor when no adapters are needed.

### PR DESCRIPTION
**Pull request by JakeWharton**

### CODE REVIEW

**Comment by JakeWharton** [comment]
No behavior change, just omits the otherwise empty constructor declaration.

File: compiler/src/main/kotlin/se/ansman/kotshi/AdaptersProcessingStep.kt [hunk 1]
```diff
@@ -110,7 +110,9 @@ class AdaptersProcessingStep(
                 .superclass(getAdapterType(typeName))
                 .addField(optionsField)
                 .addFields(generateFields(adapterKeys))
-                .addMethod(generateConstructor(adapterKeys, genericTypes))
+                .applyIf(adapterKeys.isNotEmpty()) {
```
Comment (ansman) [inline compiler/src/main/kotlin/se/ansman/kotshi/AdaptersProcessingStep.kt]: The adapter keys will be non empty when the class is a generic class, right?

Comment (JakeWharton) [reply compiler/src/main/kotlin/se/ansman/kotshi/AdaptersProcessingStep.kt]: Yes. The use of a generic requires an adapter, and elsewhere the adapter list is used as the sole source of information for pulling out that generic information (up in the factory generator I think).

Comment (ansman) [reply compiler/src/main/kotlin/se/ansman/kotshi/AdaptersProcessingStep.kt]: Cool.

**Pull request merged**